dev/com.ibm.ws.classloading/src/com/ibm/wsspi/classloading/ClassLoadingService.java

```diff
@@ -195,6 +195,16 @@ public interface ClassLoadingService {
      * @return true if an instance of ThreadContextClassLoader was provided, otherwise false.
      */
     boolean isThreadContextClassLoader(ClassLoader cl);
+    
+    /**
+     * This method returns whether or not the provided ClassLoader is a ThreadContextClassLoader
+     * and if the second provided ClassLoader is an AppClassLoader. And if the TCCL is for the AppClassLoader 
+     *
+     * @param tccl The thread context class loader object to analyze.
+     * @param appClassLoader The app class loader object to analyze.
+     * @return true if tccl is a ThreadContextClassLoader for an app classloader appClassLoader
+     */
+    boolean isThreadContextClassLoaderForAppClassLoader(ClassLoader tccl, ClassLoader appClassLoader);
```

Comment:
If we can, we should try to avoid adding new SPI in an issue like this.  At the least I think this would need some type of small design review.

Luckily the Liberty class loading service has an internal to Liberty bundles package `com.ibm.ws.classloading` which is available to other Liberty bundles.  It has the interface `com.ibm.ws.classloading.LibertyClassLoadingService<CL>` that we can add this method to.  This interface is implemented by `com.ibm.ws.classloading.internal.ClassLoadingServiceImpl` and is already provided as a service for other Liberty bundles to use.